crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -8,14 +8,27 @@ use ruma::{
 use serde::{Deserialize, Serialize};
 
 /// The verification state of the device that sent an event to us.
-#[derive(Clone, Debug, Deserialize, Serialize)]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
 pub enum VerificationState {
-    /// The device is trusted.
-    Trusted,
-    /// The device is not trusted.
-    Untrusted,
-    /// The device is not known to us.
+    /// This the only state were the authenticity is
+    /// guaranteed. It's coming from a device belonging to a
+    /// user that we have verified.
+    /// Other states give you more details about the level of trust.
+    Verified,
+    /// A signed device of an unverified user.
+    /// Message is coming from a cross signed device of a user
+    /// identity that we haven't yet verified.
+    SignedDeviceOfUnverifiedUser,
+    /// An unsigned device of a verified user.
+    UnSignedDeviceOfVerifiedUser,
+    /// The device is not signed by the user, and
+    /// we haven't verified the user
+    UnSignedDeviceOfUnverifiedUser,
```

Comment:
Updated.
We still need UnsafeSource/UnknownDevice distinction because we decorate unsafe with a lighter warning (as currently mostly noise due to legacy backup).
```
...
   None(DeviceLinkProblem),
}
```

With
```
pub enum DeviceLinkProblem {
    /// The device is deleted or unknown
    MissingDevice,
    /// The key was obtained from an insecure source.
    /// Could be imported from file, symmetric backup, unsafe forward..
    InsecureSource,
}